static/src/javascripts/lib/robust.spec.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+
+import reportError from 'lib/report-error';
+import robust from './robust';
+
+jest.mock('lib/report-error', () => jest.fn());
+
+describe('robust', () => {
+    const ERROR = new Error('Something broke.');
+    const META = { module: 'test' };
+
+    function noError() {
+        return true;
+    }
+
+    function throwError() {
+        throw ERROR;
+    }
+
+    test('catchErrorsAndLog()', () => {
+        // mock window.console.warn, to check whether it was called
+        const origConsoleWarn = window.console.warn;
+        window.console.warn = jest.fn();
+
+        expect(() => {
+            robust.catchErrorsAndLog('test', noError);
+        }).not.toThrowError();
+
+        expect(() => {
+            robust.catchErrorsAndLog('test', throwError);
+        }).not.toThrowError(ERROR);
+
+        expect(window.console.warn).toHaveBeenCalledTimes(1);
+
+        // reset window.console.warn
+        window.console.warn = origConsoleWarn;
+    });
+
+    test('catchErrorsAndLog() - default reporter', () => {
+        reportError.mockClear();
+        robust.catchErrorsAndLog('test', noError);
+        expect(reportError).not.toHaveBeenCalledWith();
```

Comment:
Are we testing to ensure that `reportError` has not been called at all? If so, I think `toHaveBeenCalled()` would work better